### ISSUE TITLE
tx-log breaks with a spec assertion fail after submitting an evict operation

### ISSUE THREAD

**Issue by danmason** [xtdb/xtdb#454]
Reproducible in `crux-dev` repl:
```clojure
(crux/submit-tx node [[:crux.tx/put {:crux.db/id :test}]])
(crux/submit-tx node [[:crux.tx/evict :test]])
(crux/tx-log node (crux/new-tx-log-context node) nil true)
```
Leads to the following spec assertion fail:
```
 Caused by clojure.lang.ExceptionInfo
   Spec assertion failed.

         Spec: :crux.api/tx-ops
        Value: [[:crux.tx/put "fb512c5da4eaa0b945f3823944a6a093eb5c42a3"]]

     Problems: 

            val: "fb512c5da4eaa0b945f3823944a6a093eb5c42a3"
             in: [0 1]
         failed: map?
           spec: :crux.api/doc
             at: [:crux.tx/put :doc]
```

**Comment by danmason** [comment]
I believe the failing spec assertion is in here (within `crux.tx`):
```clojure
(defn tx-events->tx-ops [snapshot object-store tx-events]
  (let [tx-ops (vec (for [[op id & args] tx-events]
                      (cond->> (for [arg args]
                                 (or (when (satisfies? c/IdToBuffer arg)
                                       (db/get-single-object object-store snapshot arg))
                                     arg))
                        (contains? #{:crux.tx/delete
                                     :crux.tx/evict
                                     :crux.tx/fn} op) (cons (c/new-id id))
                        true (cons op)
                        true (vec))))]
    (s/assert :crux.api/tx-ops tx-ops)
    tx-ops))
```

**Edit**

Commenting out **(s/assert :crux.api/tx-ops tx-ops)** prevents the spec assertion fail, and the following was returned from `(crux/tx-log node (crux/new-tx-log-context node) nil true)`
```
({:crux.tx/tx-id 0,
  :crux.tx/tx-time #inst "2019-12-04T15:39:37.779-00:00",
  :crux.api/tx-ops
  [[:crux.tx/put "fb512c5da4eaa0b945f3823944a6a093eb5c42a3"]]}
 {:crux.tx/tx-id 1,
  :crux.tx/tx-time #inst "2019-12-04T15:39:40.144-00:00",
  :crux.api/tx-ops
  [[:crux.tx/evict
    #crux/id "a94a8fe5ccb19ba61c4c0873d391e987982fbbd3"]]})
```

**Comment by danmason** [comment]
I suspect this may be related to https://github.com/juxt/crux/issues/364, as both issues seem to arise from the assertion check in `tx-events->tx-ops` (called within `tx-log`).

**Comment by jarohen** [comment]
closed in #464
